mathjax3-ts/input/tex/require/RequireConfiguration.ts

```diff
@@ -0,0 +1,194 @@
+/*************************************************************
+ *
+ *  Copyright (c) 2019 The MathJax Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+
+/**
+ * @fileoverview    Configuration file for the require package.
+ *
+ * @author dpvc@mathjax.org (Davide P. Cervone)
+ */
+
+import {Configuration, ConfigurationHandler} from '../Configuration.js';
+import TexParser from '../TexParser.js';
+import {CommandMap} from '../SymbolMap.js';
+import {ParseMethod} from '../Types.js';
+import TexError from '../TexError.js';
+import {TeX} from '../../tex.js';
+
+import {Package} from '../../../components/package.js';
+import {Loader, CONFIG as LOADERCONFIG} from '../../../components/loader.js';
+import {MathJax as mathjax} from '../../../mathjax.js';
+import {userOptions, OptionList, expandable} from '../../../util/Options.js';
+
+/**
+ * The MathJax configuration block (for looking up user-defined package options)
+ */
+const MJCONFIG = (global.MathJax ? global.MathJax.config || {} : {});
+
+/**
+ * Add an extension to the configuration, and configure it's user options
+ *
+ * @param {TeX} jax       The TeX jax whose configruation is to be modified
+ * @param {string} name   The name of the extension being added (e.g., '[tex]/amsCd')
+ */
+function RegisterExtension(jax: TeX<any, any, any>, name: string) {
+    const required = jax.parseOptions.options.require.required;
+    if (required.indexOf(name) < 0) {
+        const extension = name.substr(6);
+        required.push(name);
+        //
+        //  Register andy dependencies that were loaded to handle this one
+        //
+        RegisterDependencies(jax, LOADERCONFIG.dependencies[name]);
+        //
+        //  If the required file loaded an extension...
+        //
+        const handler = ConfigurationHandler.get(extension);
+        if (handler) {
+            //
+            //  Check if ther eare user-supplied options, and set those
+            //    (place them in a block for the extension, if needed)
+            //
+            let options = MJCONFIG[name];
+            if (handler.options && options) {
+                if (Object.keys(handler.options).length === 1 && handler.options[extension]) {
+                    options = {[extension]: options};
+                }
+            } else {
+                options = {};
+            }
+            //
```

Comment:
Why not just 
```
let options = MJCONFIG[name] || {};
if (handler.options) {
```
end get rid of the else case?

Reply:
The `options` variable are the user-provided options from the MathJax component configuration object. If there is no block for this extension, there is no need to create an empty sub-block for its configuration.

But I guess it doesn't hurt much to do it, so OK.